src/lib/hooks/use-query-params.ts

```diff
@@ -0,0 +1,71 @@
+'use client'
+
+import { Token } from '@indexcoop/flash-mint-sdk'
+import { useSearchParams } from 'next/navigation'
+import { useMemo } from 'react'
+
+import { getCurrencyTokens, getLeverageTokens } from '@/app/leverage/constants'
+import { LeverageToken, LeverageType } from '@/app/leverage/types'
+import { chains } from '@/lib/utils/wagmi'
+
+type DefaultParams = {
+  isMinting: boolean
+  leverageType: LeverageType
+  network: number
+  inputToken: Token
+  outputToken: Token
+}
+
+type ReturnType<T extends Partial<DefaultParams>> = {
+  [K in keyof DefaultParams as `query${Capitalize<string & K>}`]: K extends keyof T
+    ? T[K]
+    : DefaultParams[K] | undefined
+}
+
+export const useQueryParams = <T extends Partial<DefaultParams>>(
+  defaultParams: T = {} as T,
+): ReturnType<T> => {
+  const searchParams = useSearchParams()
+
+  return useMemo(() => {
+    const network = parseInt(searchParams.get('network') || '0')
```

Comment:
maybe just use `1` for undefined?

Reply:
I think dont want to force a queryparam here if its not defined.  It could be either an invalid chain, or NaN here

Reply:
So if this is not set, what happens? It just takes the default (Arbitrum) for leverage? If so, works for me.

Reply:
it not forces anything, just leave the users network